src/sqlfluff/rules/L004.py

```diff
@@ -77,8 +77,8 @@ def _eval(self, context: RuleContext) -> LintResult:
                 )
                 # Only attempt a fix at the start of a newline for now
                 and (
-                    len(context.raw_stack) == 0
-                    or context.raw_stack[-1].is_type("newline")
+                    context.raw_segment_pre is None
+                    or context.raw_segment_pre.is_type("newline")
```

Comment:
Switching from `raw_stack` to `raw_segment_pre` so this rule doesn't require  the heavyweight`raw_stack` anymore.